### PR TITLE
Replication cleanup

### PR DESCRIPTION
**Pull request by ptnapoleon**
Added in stefania's PR #688, fixed the regex, and then made other fixes so that these tests work on OSX, where nodetool calls always print to stderr

### CODE REVIEW

**Comment by ptnapoleon** [comment]
@mambocab to review, and make his style fixes

File: replication_test.py [hunk 1]
```diff
@@ -261,6 +258,11 @@ class SnitchConfigurationUpdateTest(Tester):
     """
     Test to repro CASSANDRA-10238, wherein changing snitch properties to change racks without a restart could violate RF contract.
     """
+
+    ignore_log_patterns = ["Fatal exception during initialization",
```
Comment (ptnapoleon) [inline replication_test.py]: I think you want self.ignore_log_patterns

Comment (ptnapoleon) [inline replication_test.py]: Nvm, apparently you don't

**Comment by ptnapoleon** [comment]
Please review carefully against the original PR, @mambocab, please. I missed some code in the original rebase. Tests are all passing locally now.

**Comment by mambocab** [comment]
I made sure this PR contained Stefania's changes with

```sh
git fetch datastax-gh pull/688/head:pr-688
git fetch datastax-gh pull/693/head:pr-693
git diff pr-688 pr-693
```

Looks good to me there. The changed tests pass for me locally.

**Pull request merged**